### ISSUE TITLE
imgui-sfml for bincrafters (Conan)

### ISSUE THREAD

**Issue by eliasdaler** [eliasdaler/imgui-sfml#103]
Here's the list of requirements: https://bincrafters.readthedocs.io/en/latest/contributing_to_packages/package_guidelines_faq.html
I'm not a Conan user, so everyone who uses Conan is encouraged to go through the whole process and contribute, given that we now support Conan.

**Comment by eliasdaler** [comment]
@barsoosayque - seeing how you contributed Conan PR, maybe you'll be willing to do this? :)

**Comment by barsoosayque** [comment]
Yeah, I might look into this, probably next week. 
Though, I've never contributed to bincrafters.

**Comment by barsoosayque** [comment]
I'm actually working on it, though, some compilers (the old ones) fails to build the library because of c++11 features. Is 1b8f0c776dfa0051bf45eeb17fe3b66ad1076986 remove them all (so I can create a patch based on it), or is there anything else in the repo that is also c++11 ?

**Comment by eliasdaler** [comment]
Yes, just checked - compiles fine with -std=c++98

Why create patch though, if I can release a new version which will then be updated in Conan?

**Comment by barsoosayque** [comment]
> Why create patch though, if I can release a new version which will then be updated in Conan?

Well, it would be easier this way. I just didn't want to ask for a version increment for such a minor thing.

**Comment by eliasdaler** [comment]
Let's do it this way: you'll use a patch until the final PR of updated recipe, and then I'll update the version at the same time when merging your new Conan PR into master?

**Comment by eliasdaler** [comment]
Though it might be better to use 'master' temporarily, which will be changed to a new tagged version, when merged into master (this will be easier)

**Comment by lefticus** [comment]
I'm curious what the current status is on this? Is imgui-sfml currently published to any conan repo so that it can be directly used as a requirement in a project using conan?

**Comment by barsoosayque** [comment]
I'm working on this from time to time. My working draft is here: [barsoosayque/conan-imgui-sfml](https://github.com/barsoosayque/conan-imgui-sfml), though, I can't file it for inclusion yet, because it has some CI problems. Also, there is bincrafters/community#1104 where I described my concerns about the recipe. Sadly, I don't have much time & resources to really get this working. 
So, no published recipe, but you can use my working draft as a local recipe for now.

**Comment by lefticus** [comment]
Thanks for the update

**Comment by notskm** [comment]
I've been waiting on this as well. I might take a look at it and see if I can fix the errors.

It really is a shame that the package can't depend on the imgui conan package. I ran into the same problem when I tried making an imgui-sfml package a while back.

**Comment by notskm** [comment]
I fixed it up. It should be about ready to submit. Just need to wait for CI to go green.
https://github.com/notskm/conan-imgui-sfml

**Comment by notskm** [comment]
@barsoosayque Let me know if you'd like me to submit this to Bincrafters or if you'd rather do it yourself.

**Comment by barsoosayque** [comment]
Wow, good job ! 
> Let me know if you'd like me to submit this to Bincrafters or if you'd rather do it yourself.

If it wouldn't be bothersome for you to submit it, I would like to ask you to do it. And if you do so, please notify me, so I can close that issue I mentioned above. Thanks.

**Comment by notskm** [comment]
Here's the inclusion request
bincrafters/community#1125

**Comment by eliasdaler** [comment]
Closing for now. The request has been opened and the progress for inclusion in bincrafters should be tracked in that issue.
